Analog Threshold Sensor Not Compiling

### The problem

Since updating to version 2026.3.0 the analog treshold sensor fails during compiling

### Which version of ESPHome has the issue?

2026.3.0

### What type of installation are you using?

Home Assistant Add-on

### What platform are you using?

ESP32

### Component causing the issue

analog_threshold

### YAML Config

```yaml
esphome:
  name: test
  friendly_name: Test

esp32:
  board: esp32dev
  framework:
    type: esp-idf

# Enable logging
logger:

# Enable Home Assistant API
api:
  encryption:
    key: "s0CWKIul23ewgbbFfJs8l7y8dtmneJiJvNuAOA0JNEg="

ota:
  - platform: esphome
    password: "cb616d493b5422d807d7ec8eddda491a"

wifi:
  ssid: Redacted 
  password: Redacted

  # Enable fallback hotspot (captive portal) in case wifi connection fails
  ap:
    ssid: "Test Fallback Hotspot"
    password: "motu2rrcNOjs"

sensor:
  - platform: adc
    pin: GPIO35
    name: "Bed Sensor #1"
    id: bed_sensor_1
    update_interval: 1s
#    internal: true

  - platform: adc
    pin: GPIO32
    name: "Bed Sensor #2"
    id: bed_sensor_2
    update_interval: 1s
#    internal: true  

binary_sensor:

  - platform: analog_threshold
    name: "Bed Occupancy 1"
    sensor_id: bed_sensor_1
    threshold: 
      upper: 0.8
      lower: 0.476

 

captive_portal:
```

### Anything in the logs that might be useful for us?

```txt
INFO ESPHome 2026.3.0
INFO Reading configuration /config/esphome/test.yaml...
INFO Generating C++ source...
INFO Setting CONFIG_LWIP_MAX_SOCKETS to 12 (TCP=6 [api=3, captive_portal=3], UDP=3 [captive_portal=1, mdns=2], TCP_LISTEN=3 [api=1, ota=1, web_server_base=1])
INFO Compiling app... Build path: /data/build/test
Processing test (board: esp32dev; framework: espidf; platform: https://github.com/pioarduino/platform-espressif32/releases/download/55.03.37/platform-espressif32.zip)
--------------------------------------------------------------------------------
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
 - contrib-piohome @ 3.4.4 
 - framework-espidf @ 3.50503.0 (5.5.3) 
 - tool-cmake @ 4.0.3 
 - tool-esp-rom-elfs @ 2024.10.11 
 - tool-esptoolpy @ 5.1.2 
 - tool-ninja @ 1.13.1 
 - tool-scons @ 4.40801.0 (4.8.1) 
 - toolchain-xtensa-esp-elf @ 14.2.0+20251107
Reading CMake configuration...
Dependency Graph
|-- noise-c @ 0.1.11
Compiling .pioenvs/test/src/esphome/components/analog_threshold/analog_threshold_binary_sensor.cpp.o
Compiling .pioenvs/test/src/esphome/components/binary_sensor/automation.cpp.o
Compiling .pioenvs/test/src/esphome/components/binary_sensor/binary_sensor.cpp.o
Compiling .pioenvs/test/src/esphome/components/binary_sensor/filter.cpp.o
Compiling .pioenvs/test/src/esphome/components/captive_portal/captive_portal.cpp.o
In file included from src/esphome/components/analog_threshold/analog_threshold_binary_sensor.cpp:1:
src/esphome/components/analog_threshold/analog_threshold_binary_sensor.h:21:3: error: 'TemplatableValue' does not name a type
   21 |   TemplatableValue<float> upper_threshold_{};
      |   ^~~~~~~~~~~~~~~~
src/esphome/components/analog_threshold/analog_threshold_binary_sensor.h:22:3: error: 'TemplatableValue' does not name a type
   22 |   TemplatableValue<float> lower_threshold_{};
      |   ^~~~~~~~~~~~~~~~
src/esphome/components/analog_threshold/analog_threshold_binary_sensor.h: In member function 'void esphome::analog_threshold::AnalogThresholdBinarySensor::set_upper_threshold(T)':
src/esphome/components/analog_threshold/analog_threshold_binary_sensor.h:16:76: error: 'class esphome::analog_threshold::AnalogThresholdBinarySensor' has no member named 'upper_threshold_'; did you mean 'set_upper_threshold'?
   16 |   template<typename T> void set_upper_threshold(T upper_threshold) { this->upper_threshold_ = upper_threshold; }
      |                                                                            ^~~~~~~~~~~~~~~~
      |                                                                            set_upper_threshold
src/esphome/components/analog_threshold/analog_threshold_binary_sensor.h: In member function 'void esphome::analog_threshold::AnalogThresholdBinarySensor::set_lower_threshold(T)':
src/esphome/components/analog_threshold/analog_threshold_binary_sensor.h:17:76: error: 'class esphome::analog_threshold::AnalogThresholdBinarySensor' has no member named 'lower_threshold_'; did you mean 'set_lower_threshold'?
   17 |   template<typename T> void set_lower_threshold(T lower_threshold) { this->lower_threshold_ = lower_threshold; }
      |                                                                            ^~~~~~~~~~~~~~~~
      |                                                                            set_lower_threshold
src/esphome/components/analog_threshold/analog_threshold_binary_sensor.cpp: In member function 'virtual void esphome::analog_threshold::AnalogThresholdBinarySensor::setup()':
src/esphome/components/analog_threshold/analog_threshold_binary_sensor.cpp:18:47: error: 'class esphome::analog_threshold::AnalogThresholdBinarySensor' has no member named 'lower_threshold_'; did you mean 'set_lower_threshold'?
   18 |     this->raw_state_ = sensor_value >= (this->lower_threshold_.value() + this->upper_threshold_.value()) / 2.0f;
      |                                               ^~~~~~~~~~~~~~~~
      |                                               set_lower_threshold
src/esphome/components/analog_threshold/analog_threshold_binary_sensor.cpp:18:80: error: 'class esphome::analog_threshold::AnalogThresholdBinarySensor' has no member named 'upper_threshold_'; did you mean 'set_upper_threshold'?
   18 |     this->raw_state_ = sensor_value >= (this->lower_threshold_.value() + this->upper_threshold_.value()) / 2.0f;
      |                                                                                ^~~~~~~~~~~~~~~~
      |                                                                                set_upper_threshold
src/esphome/components/analog_threshold/analog_threshold_binary_sensor.cpp: In lambda function:
src/esphome/components/analog_threshold/analog_threshold_binary_sensor.cpp:31:53: error: 'class esphome::analog_threshold::AnalogThresholdBinarySensor' has no member named 'lower_threshold_'; did you mean 'set_lower_threshold'?
   31 |           sensor_value >= (this->raw_state_ ? this->lower_threshold_.value() : this->upper_threshold_.value());
      |                                                     ^~~~~~~~~~~~~~~~
      |                                                     set_lower_threshold
src/esphome/components/analog_threshold/analog_threshold_binary_sensor.cpp:31:86: error: 'class esphome::analog_threshold::AnalogThresholdBinarySensor' has no member named 'upper_threshold_'; did you mean 'set_upper_threshold'?
   31 |           sensor_value >= (this->raw_state_ ? this->lower_threshold_.value() : this->upper_threshold_.value());
      |                                                                                      ^~~~~~~~~~~~~~~~
      |                                                                                      set_upper_threshold
In file included from src/esphome/core/component.h:10,
                 from src/esphome/components/analog_threshold/analog_threshold_binary_sensor.h:3:
src/esphome/components/analog_threshold/analog_threshold_binary_sensor.cpp: In member function 'virtual void esphome::analog_threshold::AnalogThresholdBinarySensor::dump_config()':
src/esphome/components/analog_threshold/analog_threshold_binary_sensor.cpp:43:23: error: 'class esphome::analog_threshold::AnalogThresholdBinarySensor' has no member named 'upper_threshold_'; did you mean 'set_upper_threshold'?
   43 |                 this->upper_threshold_.value(), this->lower_threshold_.value());
      |                       ^~~~~~~~~~~~~~~~
src/esphome/core/log.h:108:101: note: in definition of macro 'esph_log_config'
  108 |   ::esphome::esp_log_printf_(ESPHOME_LOG_LEVEL_CONFIG, tag, __LINE__, ESPHOME_LOG_FORMAT(format), ##__VA_ARGS__)
      |                                                                                                     ^~~~~~~~~~~
src/esphome/components/analog_threshold/analog_threshold_binary_sensor.cpp:40:3: note: in expansion of macro 'ESP_LOGCONFIG'
   40 |   ESP_LOGCONFIG(TAG,
      |   ^~~~~~~~~~~~~
src/esphome/components/analog_threshold/analog_threshold_binary_sensor.cpp:43:55: error: 'class esphome::analog_threshold::AnalogThresholdBinarySensor' has no member named 'lower_threshold_'; did you mean 'set_lower_threshold'?
   43 |                 this->upper_threshold_.value(), this->lower_threshold_.value());
      |                                                       ^~~~~~~~~~~~~~~~
src/esphome/core/log.h:108:101: note: in definition of macro 'esph_log_config'
  108 |   ::esphome::esp_log_printf_(ESPHOME_LOG_LEVEL_CONFIG, tag, __LINE__, ESPHOME_LOG_FORMAT(format), ##__VA_ARGS__)
      |                                                                                                     ^~~~~~~~~~~
src/esphome/components/analog_threshold/analog_threshold_binary_sensor.cpp:40:3: note: in expansion of macro 'ESP_LOGCONFIG'
   40 |   ESP_LOGCONFIG(TAG,
      |   ^~~~~~~~~~~~~
*** [.pioenvs/test/src/esphome/components/analog_threshold/analog_threshold_binary_sensor.cpp.o] Error 1
========================= [FAILED] Took 10.25 seconds =========================
```

### Additional information

_No response_

--- a/esphome/components/analog_threshold/analog_threshold_binary_sensor.h
+++ b/esphome/components/analog_threshold/analog_threshold_binary_sensor.h
@@ -1,5 +1,6 @@
 #pragma once
 
+#include "esphome/core/automation.h"
 #include "esphome/core/component.h"
 #include "esphome/components/binary_sensor/binary_sensor.h"
 #include "esphome/components/sensor/sensor.h"

